propolis/src/hw/nvme/mod.rs

```diff
@@ -444,12 +441,20 @@ impl PciNvme {
         let cqes = size_of::<RawCompletion>().trailing_zeros() as u8;
         let sqes = size_of::<RawSubmission>().trailing_zeros() as u8;
 
+        // Random serial number
+        let sn: [u8; 20] = rand::thread_rng()
```

Comment:
We should probably accept the serial as a parameter or generate it by seeding it off the VM's UUID or something. That way a guest won't see the same disk but with different serial number on every boot.

Reply:
Agreed.  We should probably look at exposing the volume uuid via the NGUID and/or EUI64 fields for the namespace identification.

Reply:
Unfortunately EUI64 shows up in version 1.1 and NGUID shows up in 1.2. Upgrading the supported NVMe version is a big lift to be included in this various fixes PR :)

Instead of the random serial, I plumbed through the name field out of DiskRequest:

	{
	  "name": "data",
	  "address": [
	    "127.0.0.1:3801",
	    "172.20.3.63:3801",
	    "172.20.3.60:3801"
	  ],
	  "slot": 2,
	  "read_only": false,
	  "gen": 0,
	  "key": "+7UAEsHnh8EphVql4gHE2hV6ddFXvwb1AcAlgqwe6Dk=",
	  "device": "nvme"
	}

and block device name out of the TOML:

	# extra disk
	[block_dev.extra]
	type = "file"
	path = "/home/james/disk.img"

	[dev.block1]
	driver = "pci-nvme"
	name = "Oxide Computer Company NVMe Device"
	block_dev = "extra"
	pci-path = "0.13.0"

nvme cli shows:

	root@localhost:~# ./nvme-cli/nvme list -o json
	Could not find pci.ids file
	Could not find pci.ids file
	{
	  "Devices" : [
	    {
	      "NameSpace" : 1,
	      "DevicePath" : "/dev/nvme0n1",
	      "Firmware" : "        ",
	      "Index" : 0,
	      "ModelNumber" : "                                        ",
	      "ProductName" : "NULL",
	      "SerialNumber" : "extra",
	      "UsedBytes" : 5368709120,
	      "MaximumLBA" : 10485760,
	      "PhysicalSize" : 5368709120,
	      "SectorSize" : 512
	    },
	    {
	      "NameSpace" : 1,
	      "DevicePath" : "/dev/nvme1n1",
	      "Firmware" : "        ",
	      "Index" : 1,
	      "ModelNumber" : "                                        ",
	      "ProductName" : "NULL",
	      "SerialNumber" : "data",
	      "UsedBytes" : 4288675840,
	      "MaximumLBA" : 1047040,
	      "PhysicalSize" : 4288675840,
	      "SectorSize" : 4096
	    }
	  ]
	}

That way the service issuing the DiskRequest can choose to use a name, or volume id, or something like that. See commit bd5c5d5.